src/Robo/Commands/Git/GitCommand.php

```diff
@@ -47,16 +47,42 @@ public function commitMsgHook($message) {
    * @return int
    */
   public function preCommitHook($changed_files) {
+
+
+    $changed_files_list = explode("\n", $changed_files);
+    $modified_files = [];
+
+    $extension_list = [
+      'module',
```

Comment:
I understand the value of this, but it has a negative consequence of ignoring any customizations to PHPCS  configuration that a user may have made to their phpcs.xml file.  To avoid this, we would need to load that file and parse it to determine the correct extensions, patterns, and items to ignore.

Reply:
I see, do you think you can go back to the way it's used to be before, when the PHPCS only sniffed the modified files? Is it worth investing more time in this?

Reply:
I would rather not go back to the way it was before because it required us to build and maintain our own system for managing code sniffer customizations. With the exception of this disadvantage, the current approach is much better.

I'd be open to a pull request to resolve this issue, but I did look into it myself for a bit and determined that it was not worth the time to develop.

The issue is that once you pass a list of files to phpcs, it then ignores all other settings (like include and exclude patterns).  this behavior is intentional and there's been some disagreement around it.

@see https://github.com/acquia/blt/pull/2126

> The ignoring of file extensions for specific [specified] files is a feature of PHP_CodeSniffer and is the only way to check files without an extension.

Reply:
I haven't looked into it, but we might be able to solve this by creating a custom bootstrap file: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#using-a-bootstrap-file

> These custom files may be used to perform such taks as manipulating the internal settings of PHP_CodeSniffer that are not exposed through command line arguments.